src/Symfony/Component/HttpClient/Internal/CurlClientState.php

```diff
@@ -53,7 +53,7 @@ public function __construct(int $maxHostConnections, int $maxPendingPushes)
             curl_multi_setopt($this->handle, \CURLMOPT_PIPELINING, \CURLPIPE_MULTIPLEX);
         }
         if (\defined('CURLMOPT_MAX_HOST_CONNECTIONS')) {
-            $maxHostConnections = curl_multi_setopt($this->handle, \CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : \PHP_INT_MAX) ? 0 : $maxHostConnections;
+            $maxHostConnections = curl_multi_setopt($this->handle, \CURLMOPT_MAX_HOST_CONNECTIONS, 0 < $maxHostConnections ? $maxHostConnections : \PHP_INT_MAX) ? 4294967295 : $maxHostConnections;
```

Comment:
As @nicolas-grekas mentioned, zero doesn't mean unlimited. Instead, it sets the default behavior to four times the number of added handlers. This can be seen in the [code](https://github.com/curl/curl/blob/e53523fef07894991c69d907a7c7794c7ada4ff4/lib/conncache.c#L493-L494):

```c
  unsigned int maxconnects = !data->multi->maxconnects ?
    data->multi->num_easy * 4: data->multi->maxconnects;
```

Also, since the variable is an `int`, `PHP_INT_MAX` cannot be used, as it’s too large on 64-bit systems.

Reply:
Really? curl has fixed size for `long`? Do we know if the number is signed? Aka `2147483647` would then be the max?

Reply:
@nicolas-grekas The variable is declared as `unsigned int maxconnects`. Also, I've tested it, 4294967295 works, 4294967296 doesn't any more, so it's definitely unsigned.